server/src/item.rs

```diff
@@ -168,4 +170,20 @@ impl Item {
     pub fn path(&self) -> &OwnedObjectPath {
         &self.path
     }
+
+    pub async fn set_locked(&self, locked: bool) -> Result<(), ServiceError> {
+        let manager = self.manager.lock().await;
+
+        if !self.is_locked().await {
```

Comment:
This condition implies you cant unlock an item.

Reply:
Yeah, I get it. Made a small modification to avoid this.